test/jdk/java/lang/Thread/virtual/ParkWithFixedThreadPool.java

```diff
@@ -74,4 +76,27 @@ public void run() {
             }
         }
     }
+
+    static class Scheduler implements Executor, AutoCloseable {
+        private final ExecutorService pool;
+
+        Scheduler(int poolSize) {
+            pool = Executors.newFixedThreadPool(poolSize);
+        }
+
+        @Override
+        public void execute(Runnable task) {
+            try {
+                pool.execute(task);
+            } finally {
+                // ExecutorService::execute may consume parking permit
+                LockSupport.unpark(Thread.currentThread());
```

Comment:
This seems a bit odd - why would the current thread need to unpark itself? Why should it have a park permit available here?

Reply:
In this test, Scheduler.execute method will consume the current thread's parking permit when there is contention on the queue. In a well behaved system, all usages of park will first test some condition before parking. This test doesn't do this, hence it created the scenario where parking after unparking might hang. Previous discussion in [loom/pull/59](https://github.com/openjdk/loom/pull/59). There is no support exposed for doing custom schedulers at this time but this is the type of thing that comes up so we kept the test.